graphdatascience/tests/integration/test_error_handling.py

```diff
@@ -30,7 +30,159 @@ def run_around_tests(runner: Neo4jQueryRunner) -> Generator[None, None, None]:
+        gds.graph.project.bogus("there")  # type: ignore
+
+
+def test_calling_model(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.beta.model' to call$"):
+        gds.beta.model(42, 1337)
+
+
+def test_nonexisting_model_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.beta.model.bogus' to call$"):
+        gds.beta.model.bogus(42, 1337)  # type: ignore
+
+
+def test_calling_pipeline(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.beta.pipeline' to call$"):
+        gds.beta.pipeline(42, 1337)
+
+
+def test_nonexisting_pipeline_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.beta.pipeline.bogus' to call$"):
+        gds.beta.pipeline.bogus(42, 1337)  # type: ignore
+
+
+def test_calling_linkPrediction(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.beta.pipeline.linkPrediction' to call$"):
+        gds.beta.pipeline.linkPrediction()
+
+
+def test_nonexisting_linkPrediction_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(
+        SyntaxError,
+        match="There is no 'gds.beta.pipeline.linkPrediction.whoops' to call$",
+    ):
+        gds.beta.pipeline.linkPrediction.whoops()  # type: ignore
+
+
+def test_calling_nodeClassification(gds: GraphDataScience) -> None:
+    with pytest.raises(
+        SyntaxError,
+        match="There is no 'gds.beta.pipeline.nodeClassification' to call$",
+    ):
+        gds.beta.pipeline.nodeClassification(13.37)
+
+
+def test_nonexisting_nodeClassification_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(
+        SyntaxError,
+        match="There is no 'gds.beta.pipeline.nodeClassification.whoops' to call$",
+    ):
+        gds.beta.pipeline.nodeClassification.whoops(13.37)  # type: ignore
+
+
+def test_calling_linkprediction(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.alpha.linkprediction' to call$"):
+        gds.alpha.linkprediction(1, 2, direction="REVERSE")
+
+
+def test_nonexisting_linkprediction_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.alpha.linkprediction.adamicFoobar' to call$"):
+        gds.alpha.linkprediction.adamicFoobar(1, 2, direction="REVERSE")  # type: ignore
+
+
+def test_calling_debug(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.debug' to call$"):
+        gds.debug()
+
+
+def test_nonexisting_debug_endpoint(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.debug.sniffDumpo' to call$"):
+        gds.debug.sniffDumpo()  # type: ignore
+
+
+def test_calling_util(gds: GraphDataScience) -> None:
+    with pytest.raises(SyntaxError, match="There is no 'gds.util' to call$"):
+        gds.util()
+
+
```

Comment:
I like these extensive tests, but to a large extend you are testing how well the string similarity works. I would suggest to transform some into unit tests

Reply:
(as you also pointed out, lots come from unit tests originally)

Reply:
We cannot have them as unit test, since they call `gds.list`. I think I added them in the past to make sure that they inherited the right classes, not really so much about the content of the error message

Reply:
Yes we call gds.list, but we could unit test the function that takes the list of available names and the user input.
Which kind of many of these test cases are there for.

Reply:
Hehe, well that's not really what they are for as I tried to explain. It was more a test to make sure that the correct classes, like `IllegalAttrChecker`, is inherited. Hope that makes sense :)